### DISCUSSION TITLE
Eulerian Advection Solver

### DISCUSSION THREAD

**Discussion by jbarr44** [open-atmos/PySDM#1318] [General]
Hi,

I'm just getting started with this library and am trying to get a basic 1D kinematic simulation running. One thing I'm missing is how to implement a 'solver' into the 'EulerianAdvection' dynamics. I assume this refers to a numerical integration scheme, but the documentation is not clear about what exact form the constructor is expecting. A name? A function? Would greatly appreciate clarification.

Thanks.

**Comment by slayoo** [comment]
Hello @jbarr44, thanks for giving PySDM a try!

The `EulerianAdvection` dynamic expects a callable:
https://github.com/open-atmos/PySDM/blob/91d0307a1d71bf7e03d8ae52b4a15dbe487fff21/PySDM/dynamics/eulerian_advection.py#L7-L8
https://github.com/open-atmos/PySDM/blob/91d0307a1d71bf7e03d8ae52b4a15dbe487fff21/PySDM/dynamics/eulerian_advection.py#L23

In the `Shipway_and_Hill_2012` 1D kinematic example, an `MPDATA_1D` class is defined which uses the [PyMPDATA](https://github.com/open-atmos/PyMPDATA) package for integrating single-column advection of water vapour, and is used there as argument for `EulerianAdvection` ctor:
https://github.com/open-atmos/PySDM/blob/91d0307a1d71bf7e03d8ae52b4a15dbe487fff21/examples/PySDM_examples/Shipway_and_Hill_2012/simulation.py#L49-L58
https://github.com/open-atmos/PySDM/blob/91d0307a1d71bf7e03d8ae52b4a15dbe487fff21/examples/PySDM_examples/Shipway_and_Hill_2012/simulation.py#L82

Details here:
https://github.com/open-atmos/PySDM/blob/main/examples/PySDM_examples/Shipway_and_Hill_2012/mpdata_1d.py

HTH,
Sylwester
